opam-ci-check/lib/lint_error.ml

```diff
@@ -37,9 +37,15 @@ type error =
   | PrefixConflictClassMismatch of prefix_conflict_class_mismatch
   | DefaultTagsPresent of string list
   | MissingUpperBound of string
+  | InvalidReasonForArchiving
```

Comment:
Would it be better to have the reason string in the error to make it easier to catch subtle typos?

Reply:
I did have that initially, but it added a relatively high amount of complexity to the `check_` function, and I didn't think it was worth it given this tool is currently only for use by an expert, and there are only 4 reasons to check. So, in short, I think it is not worth doing that currently, but if we add more reasons or we end up having non-experts archiving packages, we may want to add that.

Reply:
Fair enough!